api/database/mongooseIdPlugin.js

```diff
@@ -0,0 +1,7 @@
+export default function mongooseIdPlugin(schema, options) {
+  schema.options.toJSON.transform = function (doc, ret, options) {
+    delete ret._id; // Remove internal _id property before exposing it as JSON.
```

Comment:
I actually think that implementation looks fine. Sure, we could discuss a lot about validation and security, but I think that this is a trimmed down version of a real system (that is meant to be created from it), so what concerns you exactly?